go/app/main.go

```diff
@@ -1,39 +1,170 @@
 package main
 
 import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
 	"net/http"
 	"os"
 	"path"
+	"strconv"
 	"strings"
 
 	"github.com/labstack/echo/v4"
 	"github.com/labstack/echo/v4/middleware"
 	"github.com/labstack/gommon/log"
 )
 
-const (
-	ImgDir = "images"
-)
+const ImgDir = "images"
 
 type Response struct {
 	Message string `json:"message"`
 }
 
+type ItemsData struct {
+	Items []Item `json:"items"`
+}
+
+type Item struct {
+	Name     string `json:"name"`
+	Category string `json:"category"`
+	Image    string `json:"image"`
+}
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
+func errMessage(c echo.Context, err error, status int, message string) error {
+	errorMessage := fmt.Sprintf("%s:%s", message, err)
+	return c.JSON(status, Response{Message: errorMessage})
+
+}
+
+func readFile(c echo.Context, filePath string) (ItemsData, error) {
+	var items ItemsData
+	file, err := os.Open(filePath)
+	if err != nil {
+		return items, errMessage(c, err, http.StatusBadRequest, "Unable to open the file")
+	}
+	defer file.Close()
+	jsonData, err := ioutil.ReadAll(file)
+	if err != nil {
+		return items, errMessage(c, err, http.StatusBadRequest, "Unable to read json data")
+	}
+	err = json.Unmarshal(jsonData, &items)
+	if err != nil {
+		return items, errMessage(c, err, http.StatusBadRequest, "Unable to unmarshal")
+	}
+	return items, nil
+}
+
 func addItem(c echo.Context) error {
+	var res Response
+	var items []Item
+	var itemsData ItemsData
+
 	// Get form data
 	name := c.FormValue("name")
-	c.Logger().Infof("Receive item: %s", name)
+	category := c.FormValue("category")
+	image, err := c.FormFile("image")
+	if err != nil {
+		errMessage(c, err, http.StatusBadRequest, "Unable to get image")
+	}
+
+	imageName, err := hashFile(c, image)
```

Comment:
Nice job reading and hashing the file. Now you just need to save the file to `ImgDir`.